usaspending_api/search/v2/views/spending_over_time.py

```diff
@@ -89,31 +102,107 @@ def database_data_layer(self):
+
+        if self.group == "quarter":
+            time_period["quarter"] = str(key_as_date.quarter)
+        elif self.group == "month":
+            time_period["month"] = str(key_as_date.month)
+
+        aggregated_amount = bucket.get("sum_as_dollars", {"value": 0})["value"]
+        return {"aggregated_amount": aggregated_amount, "time_period": time_period}
+
+    def build_elasticsearch_result(self, agg_response: AggResponse, time_periods: list) -> list:
+        results = []
+        min_date, max_date = min_and_max_from_date_ranges(time_periods)
+        date_range = generate_fiscal_date_range(min_date, max_date, self.group)
+        date_buckets = agg_response.group_by_time_period.buckets
+        parsed_bucket = None
+
+        for date in date_range:
+            if date_buckets and parsed_bucket is None:
+                parsed_bucket = self.parse_elasticsearch_bucket(date_buckets.pop(0))
+
+            time_period = {"fiscal_year": str(date.year)}
+            if self.group == "quarter":
+                time_period["quarter"] = str(date.quarter)
+            elif self.group == "month":
+                time_period["month"] = str(date.month)
+
+            if parsed_bucket is not None and time_period == parsed_bucket["time_period"]:
+                results.append(parsed_bucket)
+                parsed_bucket = None
+            else:
+                results.append({"aggregated_amount": 0, "time_period": time_period})
+
+        return results
+
+    def query_elasticsearch(self, time_periods: list) -> list:
+        search = get_base_search_with_filters(
+            index_name="{}*".format(settings.ES_TRANSACTIONS_QUERY_ALIAS_PREFIX), filters=self.filters
+        )
+        self.apply_elasticsearch_aggregations(search)
+        response = search.execute()
+        return self.build_elasticsearch_result(response.aggs, time_periods)
+
     @cache_response()
-    def post(self, request):
+    def post(self, request: Request) -> Response:
         json_request = self.validate_request_data(request.data)
-        self.group = json_request["group"]
+        self.group = GROUPING_LOOKUP[json_request["group"]]
         self.subawards = json_request["subawards"]
         self.filters = json_request["filters"]
-
-        db_results, values = self.database_data_layer()
+        self.elasticsearch = is_experimental_elasticsearch_api(request)
 
         # time_period is optional so we're setting a default window from API_SEARCH_MIN_DATE to end of the current FY.
         # Otherwise, users will see blank results for years
         current_fy = generate_fiscal_year(datetime.now(timezone.utc))
-        if self.groupings[self.group] == "fiscal_year":
+        if self.group == "fiscal_year":
             end_date = "{}-09-30".format(current_fy)
         else:
-            end_date = "{}-{}-30".format(current_fy, datetime.now(timezone.utc).month)
+            current_month = datetime.now(timezone.utc).month
```

Comment:
Any issues with the end date year being the fiscal year but the month as the calendar month?

Reply:
I will make a change here. To keep consistent with how we convert there searched dates over to FY this should be changed.